test/performance/shmem_perf_suite/uni_dir.h

```diff
@@ -56,3 +60,38 @@ void inline uni_dir_bw(int len, perf_metrics_t *metric_info)
         calc_and_print_results((end - start), len, *metric_info);
     }
 }
+
+void static inline uni_bw_get(int len, perf_metrics_t *metric_info)
+{
+    double start = 0.0, end = 0.0;
+    int i = 0, j = 0;
+    int dest = partner_node(*metric_info);
+    int snode = (metric_info->num_pes != 1)? streaming_node(*metric_info) : true;
+
+    if(metric_info->target_data) {
+        target_bw_itr(len, metric_info);
+        return;
+    }
+
+    shmem_barrier_all();
+
+    if (snode) {
+        for (i = 0; i < metric_info->trials + metric_info->warmup; i++) {
+            if(i == metric_info->warmup)
+                start = perf_shmemx_wtime();
+
+            for(j = 0; j < metric_info->window_size; j++) {
+#ifdef USE_NONBLOCKING_API
+                shmem_getmem_nbi(metric_info->dest, metric_info->src, len, dest);
+                shmem_quiet();
```

Comment:
Do we need a `quiet` for each `getmem_nbi`, or can we get away with one after the loop?  Looks to me like blocking and non-blocking will achieve the exact same performance with all the quiets.  However, each get has the same destination... so maybe this is intended?

Reply:
This may be an artifact of the contexts prototype.  I think we only prototyped the nonblocking API, so to get blocking behavior in the benchmark we had to do this.  So, the quiet should probably be deleted.

Reply:
@jdinan Do you mean that for non-blocking also, quiet is not needed?

Reply:
We should not need to invoke quiet after each operation.  It doesn't look like we're consistent right now about where it's done (per operation, per window, etc.).  Technically what we're doing is undefined by the OpenSHMEM spec (both blocking and nonblocking cases), since each put/get operation is writing to the same location.  But, I think we chose to cross our fingers and ignore that a while back (a comment to document this wouldn't hurt).

Reply:
Yes, the fact that we are writing in the same location every iteration is semantically different what a real application would do with these APIs. However, performance benchmarking has always been this way, I guess. Alternatively, should we create a series of buffers and write in different locations in each iteration instead? That may require some changes in all the benchmarks but will reflect real application behavior.

Reply:
For now, I think a comment should suffice.

Reply:
ok, thanks

Reply:
The quiet in the latency test is probably correct.